### PR TITLE
Docs/client search with js search

### PR DESCRIPTION
**Pull request by jonniebigodes**
Revised draft of the documentation of client side search with js-search and gatsby with the comments by @m-allanson and @shannonbux incorporated. 

accompanying example was rebased to the hello-world-starter.

### CODE REVIEW

File: examples/using-js-search/src/components/ClientSearch.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    if (selectedStrategy === 'Exact match') {
+      dataToSearch.indexStrategy = new JsSearch.ExactWordIndexStrategy()
+    }
+    if (selectedStrategy === 'Prefix match') {
+      dataToSearch.indexStrategy = new JsSearch.PrefixIndexStrategy()
+    }
+    selectedSanitizer === 'Case Sensitive'
+      ? (dataToSearch.sanitizer = new JsSearch.CaseSensitiveSanitizer())
+      : (dataToSearch.sanitizer = new JsSearch.LowerCaseSanitizer())
+    termFrequency === true
```
Comment (m-allanson) [inline examples/using-js-search/src/components/ClientSearch.js]: eslint complains about this and the previous statement. You could restructure the statements to something like:

```js
    dataToSearch.searchIndex =
      termFrequency === true
        ? new JsSearch.TfIdfSearchIndex(`isbn`)
        : new JsSearch.UnorderedSearchIndex()
```

or just disable eslint for these lines with `// eslint-disable-next-line no-unused-expressions`

Comment (jonniebigodes) [reply examples/using-js-search/src/components/ClientSearch.js]: the example files were not linted on purpose. I was aiming for a review on the documentation first, then lint the files and commit them

**Comment by m-allanson** [comment]
I think this is looking good! 👍 I made some edits to the docs page, mostly small fixes and shuffling content around to try and improve the document flow. 

I feel like the example site is much easier to follow now with the reduced amount of code. If you can fix up the lint errors then I think this will be ready to merge.

**Comment by jonniebigodes** [comment]
@m-allanson doing the final check on the files. Tomorrow i'll make a final check and commit them, so that we can merge this and continue with the documentation expansion. Sounds good?

**Comment by m-allanson** [comment]
@jonniebigodes Sounds good!

**Comment by LekoArts** [comment]
Would it be possible to make a graphql query to e.g. the blog sites you have and put that into the searchIndex instead of grabbing a JSON file from the internet?

I think that would be more what people would want to do.

**Comment by jonniebigodes** [comment]
@LekoArts this is the most basic example possible, in order to demonstrate that basic options of the library. And the example has been rebased and trimmed down to avoid further confusion.
What you mentioned in the comment is probably possible, but the scope of this example/document is how to create the most simple client side search possible

**Comment by LekoArts** [comment]
I asked the question because someone else asked that on Discord, and I personally think that he has a point. I'd say it’s more common to want to index e.g. the blogposts you already have. In your example the people would have to build the index themselves, right?

**Comment by jonniebigodes** [comment]
Let's start from here and then expand the documentation if necessary

**Comment by jonniebigodes** [comment]
@m-allanson is there anything more needed from me? or can this be merged and closed?
@LekoArts i've got a quasi example ready based on the comment you added. I'm going to make some more tweaks to it, if nothing pops up in real life, i'll have it ready mid week, otherwise during the weekend i'll go over it and some adjustments. Sounds good?

**Comment by m-allanson** [comment]
@jonniebigodes it looks like it's just the lint check that is causing some warnings. You should be able to fix them by running `yarn run format` and then pushing the changes up to your branch.

**Comment by jonniebigodes** [comment]
@m-allanson hummm...ok...i just did that and i'm now presented with this. 

If i recall, it happened last time aswell, i did not commit these files due to the fact that they were unrelated to the pull request.
error_component

And
final_outcome

Are these 24 files needed?

**Comment by m-allanson** [comment]
oh hmm that's odd - I'll try it out on this side to see what's up.

**Comment by jonniebigodes** [comment]
No need to thank, happy to help. And if it liked it that way it's fine by me. Going to close and delete the branch to avoid cluttering and now i can have lunch 😁

**Pull request merged**